controllers/operatorconfig_controller.go

```diff
@@ -54,51 +76,343 @@ type OperatorConfigReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
+	// Fetch the OperatorConfig operatorConfig
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, req.NamespacedName, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 
-	err := createClusterTasks(ctx, log)
+	if operatorConfig.Name != OwnerConfigName {
+		reason := fmt.Sprintf("Invalid name (%s)", operatorConfig.Name)
+		msg := fmt.Sprintf("Only one OperatorConfig supported per cluster and must be named '%s'", OwnerConfigName)
+		log.Info(fmt.Sprintf("%s: %s", reason, msg))
+		meta.SetStatusCondition(&operatorConfig.Status.Conditions, metav1.Condition{
+			Type:               InvalidConfigName,
+			Status:             True,
+			Reason:             "OperatorConfigNameInvalid",
+			Message:            fmt.Sprintf("%s: %s", reason, msg),
+			LastTransitionTime: metav1.Time{Time: time.Now()},
+		})
+		err := r.Status().Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
```

Comment:
nit: do we need to apply the same configuration but turn it to False here?

```
		meta.SetStatusCondition(&operatorConfig.Status.Conditions, metav1.Condition{
			Type:               InvalidConfigName,
			Status:             False,
			Reason:             "StandardNameFound",
			Message:            fmt.Sprintf("%s: %s", reason, msg),
			LastTransitionTime: metav1.Time{Time: time.Now()},
		})
```

Reason why I ask is as follows:

> Condition status values may be True, False, or Unknown. The absence of a condition should be interpreted the same as Unknown. How controllers handle Unknown depends on the Condition in question.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions